docs/transactions.txt

```diff
@@ -9,71 +9,148 @@ Transactions
+.. literalinclude:: /includes/fundamentals/transactions/TransactionsTest.php
+   :language: php
+   :dedent:
+   :start-after: begin transaction callback
+   :end-before: end transaction callback
 
-To abort a transaction, call the ``rollBack`` method at any point during the transaction:
+You can optionally pass the maximum number of times to retry a failed transaction as the second parameter as shown in the following code example:
 
 .. code-block:: php
+   :emphasize-lines: 4
 
-   DB::beginTransaction();
-   User::create(['name' => 'john', 'age' => 19, 'title' => 'admin', 'email' => 'john@example.com']);
+   DB::transaction(function() {
+           // transaction code
+       },
+       retries: 5,
+   );
 
-   // Abort the transaction, discarding any data created as part of it
-   DB::rollBack();
+.. _laravel-transaction-commit:
 
+Begin and Commit a Transaction
+------------------------------
 
-.. note::
+This section shows how to start and commit a transaction.
 
-   Transactions in MongoDB cannot be nested. DB::beginTransaction() function 
-   will start new transactions in a new created or existing session and will
-   raise the RuntimeException when transactions already exist. See more in 
-   MongoDB official docs :manual:`Transactions and Sessions </core/transactions/#transactions-and-sessions>`.
+To use this method of starting and committing a transaction, call the
+``DB::beginTransaction()`` to start the transaction. Then, call the
+``DB::commit()`` method to end the transaction, which applies all the updates
+performed within the transaction.
 
-.. code-block:: php
+In the following example, the balance from the first account is moved to the
+second account, after which the first account is deleted:
 
-   DB::beginTransaction();
-   User::create(['name' => 'john', 'age' => 20, 'title' => 'admin']);
+.. literalinclude:: /includes/fundamentals/transactions/TransactionsTest.php
+   :language: php
+   :dedent:
+   :emphasize-lines: 1,9
+   :start-after: begin commit transaction
+   :end-before: end commit transaction
 
-   // This call to start a nested transaction will raise a RuntimeException
-   DB::beginTransaction();
-   DB::collection('users')->where('name', 'john')->update(['age' => 20]);
-   DB::commit();
-   DB::rollBack();
+.. _laravel-transaction-rollback:
 
-Database Testing
-----------------
+Roll Back a Transaction
+-----------------------
 
-For testing, the traits ``Illuminate\Foundation\Testing\DatabaseTransactions``
-and ``Illuminate\Foundation\Testing\RefreshDatabase`` are not yet supported.
-Instead, create migrations and use the ``DatabaseMigrations`` trait to reset 
-the database after each test:
+This section shows how to roll back a transaction. A rollback reverts all the
+write operations performed within that transaction. This means that the
+data is reverted to its state before the transaction.
 
-.. code-block:: php
+To perform the roll back, call the ``DB::rollback()`` function anytime before
```

Comment:
S: I think it makes sense to use "roll back" as the verb form and "rollback" as the noun form
```suggestion
To perform the rollback, call the ``DB::rollback()`` function anytime before
```